### ISSUE TITLE
Document implementation details on the Javascript client

### ISSUE THREAD

**Issue by pcplpc** [spring-epfl/lightnion#7]
Few details have been given in answer of https://github.com/spring-epfl/lighttor/commit/5e29d15fcd67f72e2fd81cf6d93f1889a0619893 and should be integrated with the documentation of the library – as GitHub comments are not a practical way of documenting the library in a long-term, resilient fashion.

This is especially interesting as most of the library have been designed to have all of its parts at least configurable and for the most part completely replaceable by a custom implementation of the appropriate handler – this will be useful only if people can hack into it with well-documented behaviors.

**TL;DR:** Giving more insights on the implementation details of the client will enable fellow power-users and researchers to hack their way through the code.
